### PR TITLE
Uses new Azure.Provisioning packages which have been split by resource

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR (draft until we get a `0.1.0-beta.x build`) incorporates a change that the Azure SDK team has made around the `Azure.Provisioning` package. They are breaking it up into multiple packages. Here is what is in this PR:

# Migration to Azure.Provisioning.* packages

The Azure SDK team has split up the `Azure.Provisioning` packages into mutliple packages based on resource. This is something that we had always planned on happening as it better aligns with the way their engineering system works. We held off doing it for P5 due to the tight timeframe but it is now ready for P6.

# Fix for AppInsights/LogAnalytics issue

Through some manual testing I found an edge case that I hadn't tested earlier in relation to integration of log analytics and app insights. If someone does not provide a log analytics workspace explicitly then we were not creating one (we used to with our Bicep based approach). This was not an issue for deployment with AZD because it would inject the log analytics workspace ID. However if you don't provide a workspace AND you use the Azure Provisioner the deployment would fail. This PR includes fixes for that issue as well as covering test cases so that we don't accidentally regress this behavior.

# Firewall rules

Reacted to some changes in the firewall configuration for resources via the CDK. Unfortunately we won't be able to add an IP based firewall for KeyVault, Storage, and Cosmos resources. For KeyVault and Storage - they don't treat ACA as a trusted service (probably got something to do with the highly ephemeral IP ranges for this service). For Cosmos you can enable the firewall, but upon inspection of the side effects of doing this manually in the Portal it appears that a set of hard coded IP values are added. This is probably not something that we want to do in Aspire so we'll be better off waiting until we figure out how to make Aspire work well with virutal networks.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3383)

### CODE REVIEW

**Comment by mitchdenny** [comment]
@joperezr @danmoseley ... heads up for an incoming change for P6/GA.

File: NuGet.config
Comment (eerhardt) [inline NuGet.config:22]: When will these be available on nuget.org? Doing this isn't keeping us in a "shippable state" and we will need to remove this feed and update the versions before we release preview6.

File: src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs:52]: > Unfortunately 

Is there some sort of issue tracking this? Do we need to raise this problem somewhere?

File: playground/AspireEventHub/EventHubs.AppHost/eventhubns.module.bicep
Comment (eerhardt) [inline playground/AspireEventHub/EventHubs.AppHost/eventhubns.module.bicep:26]: Where is this change coming from? Is 1.2 the default and we don't need to explicitly specify it in the bicep?

Also minor nit - why is it generating an empty `properties` object. Couldn't that just be omitted?

File: playground/AspireEventHub/EventHubs.AppHost/ehstorage.module.bicep
Comment (eerhardt) [inline playground/AspireEventHub/EventHubs.AppHost/ehstorage.module.bicep:13]: Do these names change _every time_ the bicep is generated? If so, how do we expect people to do something like infra synth and easily diff when there are so many diffs.

File: playground/ProxylessEndToEnd/ProxylessEndToEnd.AppHost/aspire-manifest.json
Comment (eerhardt) [inline playground/ProxylessEndToEnd/ProxylessEndToEnd.AppHost/aspire-manifest.json:12]: I assume/know this isn't caused by your change, but are these ports supposed to go into the manifest?

File: src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs
Comment (eerhardt) [inline src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs:87]: Is this correct? Is it supposed to be:

```suggestion
                autoInjectedLogAnalyticsWorkspace.Properties.Tags["aspire-resource-name"] = $"law-{construct.Resource.Name}";
```

Also, can we remove the duplicate `$"law-{construct.Resource.Name}"` and just use `autoInjectedLogAnalyticsWorkspace.Name` to reuse the value we set on line 86?

File: NuGet.config
Comment (mitchdenny) [reply NuGet.config:22]: This will be removed before the PR merges. Working through this with the Azure SDK team. This was just to get it working and validated so we can greenlight that.

File: playground/AspireEventHub/EventHubs.AppHost/eventhubns.module.bicep
Comment (mitchdenny) [reply playground/AspireEventHub/EventHubs.AppHost/eventhubns.module.bicep:26]: 1.2 would be the default.

The empty properties field might be necessary to satisfy the RP.

Letting @JoshLove-msft know about the change here.

File: playground/AspireEventHub/EventHubs.AppHost/ehstorage.module.bicep
Comment (mitchdenny) [reply playground/AspireEventHub/EventHubs.AppHost/ehstorage.module.bicep:13]: We changed the formula we use for resource name generation deliberately to address some linter warnings. This bicep was first generated using the old method and had not been updated until I regenerated everything.

These should be stable moving forward. It's a good call out though as we cannot change these moving forward.

File: playground/ProxylessEndToEnd/ProxylessEndToEnd.AppHost/aspire-manifest.json
Comment (mitchdenny) [reply playground/ProxylessEndToEnd/ProxylessEndToEnd.AppHost/aspire-manifest.json:12]: Yes. I just happen to be the one that regenerates all the manifests ;)

File: src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs
Comment (mitchdenny) [reply src/Aspire.Hosting.Azure.ApplicationInsights/AzureApplicationInsightsExtensions.cs]: That name field is the bicep construct name. And the other one would be an expression. 3 different values. But I can tweak this a bit so I only generate the string once.

File: Directory.Packages.props
Comment (davidfowl) [inline Directory.Packages.props:51]: Why not use a variable here?

File: eng/refreshManifests.ps1
Comment (davidfowl) [inline eng/refreshManifests.ps1:1]: Why does it build first?

File: Directory.Packages.props
Comment (mitchdenny) [reply Directory.Packages.props:51]: Because they will eventually start releasing at a different cadence. Right now these package are all in the sdk/provisioning folder but they will end up splitting them out into sdk/[service] folders and will tend to release on a cadence tied to the underlying management plane packages.

File: eng/refreshManifests.ps1
Comment (mitchdenny) [reply eng/refreshManifests.ps1:1]: To make sure that everything is up to date because the loop uses --no-build (to speed things up).

File: Directory.Packages.props
Comment (davidfowl) [reply Directory.Packages.props:51]: Can't we change it when that happens?

File: eng/refreshManifests.ps1
Comment (davidfowl) [reply eng/refreshManifests.ps1:1]: This shouldn't run build.cmd. Let users run that.

**Comment by davidfowl** [comment]
As part of these firewall rule changes, these services still all work when using your local dev machine right? Those rules only apply when the compute is in azure?

**Comment by mitchdenny** [comment]
Correct. I've been running the compute locally and using the services.

**Comment by mitchdenny** [comment]
@davidfowl we need to get this into main.

**Review by davidfowl — APPROVED** [review]

**Comment by davidfowl** [comment]
Lets merge it!

**Pull request merged**
